test.sh

```diff
@@ -1,5 +1,67 @@
 #!/bin/bash
 
 # main test runner for heketi
+# Executes all executable scripts under tests dir
+# in sorted order.
```

Comment:
Why does the ordering of the tests matter? In fact why not remove the `123-` prefix anyways?

Reply:
Thanks. I wanted to be able to run the faster, simpler checks first before moving on to items of more detail/slower. This has the added benefit that when used with the -x/--exitfirst option that simple checks, such as go vet for example, run before the test suite and if it fails the suite stops.

Reply:
@phlogistonjohn Agreed--- you should do that this way IMO: https://github.com/purpleidea/mgmt/blob/master/test.sh#L20

Reply:
I see pros and cons:
* The downside is that you have to manually adapt the list scripts to be executed while @phlogistonjohn 's approach automatically executes them when they get added.
* The upside is that tests are not run automatically if they are placed but only if they are explicitly added to the runner script.
Not quite sure which I like best.

Also please take a look at the tests infra in https://github.com/gluster/gluster-kubernetes/tree/master/tests/simple which also has a traversal mechansim, and uses a subunit formatter for outputting the test results. Just to add to the confusion. ;-)

Reply:
Unsurprisingly, I like the current approach. :-)
FWIW I did look at the traversal and subunit stuff in there. At this time it seemed a bit big for what I was trying to accomplish in these initial changes. But I'm willing to down that road if you think it best.

Reply:
no need imho. you got a point - keep it simple for a start. we can still add more stuff later...

Reply:
Actually the core scripts which don't change much (if ever) are listed in order statically in mgmt, whereas all the individual integration tests (which don't need to run in a particular order) are dumped into a folder.

One problem with the numbering system is if you ever need to "make room" or adjust things, moving them around will mess up the commit history, making it harder to see what happened.

Reply:
That's a fair point. I'd like @obnoxxx to weigh in on if I should do that or stick with my original approach.

Reply:
We could work with symlinks:

```
001.sh --> this_is_the_real_test_script_name.sh
002.sh --> name_of_the_script_never_needs_to_change.sh
...
```

:-D

Seriously, I don't care too much. I even don't care too much about the order. You can see in gluster-kubernetes that I just picked up all files in the dir without particular order (apart from the name). In samba, in the ctdb-sub-component, we use numbered scripts, and that has not created too much of a disturbance.

So I won't NACK any approach. I'd vote to keep it as simple as possible.

Reply:
(we can always change it later on...)

Reply:
I like the model proposed by @phlogistonjohn here. Nothing stops us from having two tests with same NNN if we are comfortable with either running first.

Reply:
To explain further, instead of considering the number at the start as a exact place in linear order, consider them as a run level. We can have 100 run levels and nothing stops us from having 10-15 tests in each run level, I can have 001-testthis.go and 001-testthat.go iff we don't care which is run first. @purpleidea @phlogistonjohn @obnoxxx agree?

Reply:
That's a yes from me.